doc/source/whatsnew/v1.5.0.rst

```diff
@@ -1065,7 +1065,7 @@ Other
 .. ***DO NOT USE THIS SECTION***
 
 - Bug in :func:`.assert_index_equal` with ``names=True`` and ``check_order=False`` not checking names (:issue:`47328`)
--
+- Bug in :meth:`DataFrame.eval` when pandas objects (e.g. ``'Timestamp'``) were column names (:issue:`44603`)
```

Comment:
@weikhor I pushed a change to fix the wording.

Could you move this note under the `Conversion` header?